code/modules/projectiles/pins.dm

```diff
@@ -53,8 +52,8 @@
 	gun = null
 	return
 
-/obj/item/firing_pin/proc/pin_auth(mob/living/user)
-	return 1
+/obj/item/firing_pin/proc/pin_auth(mob/living/user)			//User can be null, do an istype first.
```

Comment:
Assume the type is correct and let it runtime if someone fucks up, this isn't the sort of proc that needs to be immune to incorrect types. Just check that the user var contains anything at all, not an istype.